File: basic recursion/javascript/sum_of_first_n.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function sumOfFirstN(n) {
    if (n <= 0) {
        return 0;
    }
    return n + sumOfFirstN(n - 1);
}

rl.question("Enter a number: ", function(n) {
    n = parseInt(n);
    console.log("Sum of first", n, "natural numbers is:", sumOfFirstN(n));
    rl.close();
});

File: basic recursion/c++/sum_of_first_n.cpp
#include <iostream>
using namespace std;

int sumOfFirstN(int n)
{
    if (n == 0)
    {
        return 0;
    }
    return n + sumOfFirstN(n - 1);
}
int main()
{
    int n;
    cout << "Enter a number: ";
    cin >> n;
    cout << "Sum of first " << n << " natural numbers is: " << sumOfFirstN(n) << endl;
    return 0;
}